extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosInterface.java

```diff
@@ -186,16 +190,14 @@ BatchMediaItemResponse createPhoto(NewMediaItemUpload newMediaItemUpload) throws
       // if the response is "unauthorized", refresh the token and try the request again
       if (e.getStatusCode() == 401) {
         monitor.info(() -> "Attempting to refresh authorization token");
-        if (credential.refreshToken()) {
-          // if the second attempt throws an error, then something else is wrong, and we bubble up the response errors
-          monitor.info(() -> "Refreshed authorization token successfuly");
-          response = requestFactory
-              .buildPostRequest(new GenericUrl(url + "?" + generateParamsString(parameters)),
-                  httpContent).execute();
-          
-        } else {
-          throw new IOException("Couldn't refresh authorization token after retrying");
-        }
+        // if the credential refresh failed, let the error bubble up via the IOException that gets thrown
```

Comment:
IOExceptions are the ones that get swallowed, so I'm not sure this will get bubbled up as far as you expect it to, depending on whether this code is getting called from within a ExecuteAndSwallowErrors stack

Reply:
In the logs ive seen them bubbled up and retried - which is expected ex: 
```
Caught exception java.io.IOException: Couldn't refresh authorization token after retrying at org.datatransferproject.datatransfer.google.photos.GooglePhotosInterface.makePostRequest(GooglePhotosInterface.java:197) at org.datatransferproject.datatransfer.google.photos.GooglePhotosInterface.listMediaItems(GooglePhotosInterface.java:112) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.exportPhotos(GooglePhotosExporter.java:199) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.export(GooglePhotosExporter.java:129) at org.datatransferproject.datatransfer.google.photos.GooglePhotosExporter.export(GooglePhotosExporter.java:58) at org.datatransferproject.transfer.CallableExporter.call(CallableExporter.java:62) at org.datatransferproject.transfer.CallableExporter.call(CallableExporter.java:36) at org.datatransferproject.types.transfer.retry.RetryingCallable.call(RetryingCallable.java:66) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:137) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copyHelper(PortabilityInMemoryDataCopier.java:194) at org.datatransferproject.transfer.PortabilityInMemoryDataCopier.copy(PortabilityInMemoryDataCopier.java:95) at org.datatransferproject.transfer.JobProcessor.processJob(JobProcessor.java:122) at org.datatransferproject.transfer.Worker.doWork(Worker.java:39) at org.datatransferproject.transfer.WorkerMain.poll(WorkerMain.java:138) at org.datatransferproject.transfer.WorkerMain.main(WorkerMain.java:64)
```

but any errors that happened during the call to refresh the credential from within the credential library got swallowed (it just returns a Boolean for success) so we have no insight into why it failed.